packages/rs-drive-abci/src/execution/platform_events/withdrawals/mark_chainlocked_withdrawals_as_complete/v0/mod.rs

```diff
@@ -0,0 +1,311 @@
+use dashcore_rpc::json::AssetUnlockStatus;
+use dpp::block::block_info::BlockInfo;
+use dpp::data_contract::accessors::v0::DataContractV0Getters;
+use dpp::data_contracts::withdrawals_contract::WithdrawalStatus;
+use dpp::document::document_methods::DocumentMethodsV0;
+use dpp::document::{DocumentV0Getters, DocumentV0Setters};
+use dpp::platform_value::btreemap_extensions::BTreeValueMapHelper;
+use dpp::system_data_contracts::withdrawals_contract::v1::document_types::withdrawal;
+use dpp::version::PlatformVersion;
+use itertools::Itertools;
+use std::collections::HashSet;
+
+use drive::drive::batch::DriveOperation;
+use drive::drive::identity::withdrawals::WithdrawalTransactionIndex;
+use drive::grovedb::TransactionArg;
+
+use crate::execution::types::block_execution_context::v0::BlockExecutionContextV0Getters;
+use crate::execution::types::block_state_info::v0::BlockStateInfoV0Getters;
+use crate::platform_types::epoch_info::v0::EpochInfoV0Getters;
+use crate::{
+    error::{execution::ExecutionError, Error},
+    platform_types::platform::Platform,
+    rpc::core::CoreRPCLike,
+};
+
+impl<C> Platform<C>
+where
+    C: CoreRPCLike,
+{
+    /// Update statuses for broadcasted withdrawals
+    pub(super) fn mark_chainlocked_withdrawals_as_complete_v0(
+        &self,
+        block_info: &BlockInfo,
+        transaction: TransactionArg,
+        platform_version: &PlatformVersion,
+    ) -> Result<(), Error> {
+        let mut broadcasted_withdrawal_documents =
+            self.drive.fetch_withdrawal_documents_by_status(
```

Comment:
If we have many lost txs in broadcasted state we will fetch them everytime and everytime check if transaction is mined (forever) that will make this function very inefficient